src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -168,6 +171,22 @@ public DocumentDatabase(string name, RavenConfiguration configuration, ServerSto
                 });
                 _hasClusterTransaction = new AsyncManualResetEvent(DatabaseShutdown);
                 IdentityPartsSeparator = '/';
+
+                switch (serverStore.Configuration.Indexing.JournalSynchronizationMode)
+                {
+                    case IndexWriteSynchronizationMode.Server:
+                        this.IndexWritesSynchronization = serverStore.IndexWritesSynchronization;
+                        break;
+                    case IndexWriteSynchronizationMode.Database:
+                        this.IndexWritesSynchronization = new StorageEnvironmentSynchronization(
+                                  serverStore.Configuration.Indexing.JournalMaxConcurrentWrites,
+                                  serverStore.Configuration.Indexing.JournalMaxConcurrentWritesSizeInMegabytes * Sparrow.Global.Constants.Size.Megabyte,
```

Comment:
use `Size` type as a parameter, we do not want to deal with conversions manually. You will get it from the configuration after applying my previous remark.